xwiki-platform-core/xwiki-platform-user/xwiki-platform-user-profile/xwiki-platform-user-profile-ui/src/main/resources/XWiki/XWikiUserPreferencesSheet.xml

```diff
@@ -247,6 +389,50 @@
   #end
 #end
 ###############################
+##   DISPLAY SHORTCUT PREFERENCE
+###############################
+#macro (displayShortcut $fieldName $label $translationKey)
+  #set ($hintKey = "XWiki.UserPreferenceShortcutsClass_${fieldName}.hint")
+  #set ($escapedSelectId = $escapetool.xml("XWiki.UserPreferenceShortcutsClass_${fieldNumber}"))
+  #set ($escapedHintId = $escapedSelectId + "_hint")
+  #set ($defaultValue = $escapetool.xml($services.localization.render($translationKey)))
+  &lt;dt&gt;
+    &lt;label #if($isEdit)for="$escapedSelectId"#end&gt;$!escapetool.xml($services.localization.render($label))&lt;/label&gt;
+    #if ($services.localization.get($hintKey))
+      &lt;span id="$escapedHintId" class='xHint'&gt;$!escapetool.xml($services.localization.render($hintKey)) ##
+      $escapetool.xml($services.localization.render('userprofile.default')): $defaultValue##
```

Comment:
The default value should be put as placeholder inside the `userprofile.default` key (which also seems a bit short/non-specific).

Reply:
I've been following the pattern already used in https://github.com/xwiki/xwiki-platform/blob/6bb5cf74c28b0fe12e29117aaa3f1f39a7b9ded1/xwiki-platform-core/xwiki-platform-user/xwiki-platform-user-profile/xwiki-platform-user-profile-ui/src/main/resources/XWiki/XWikiUserPreferencesSheet.xml#L356-L375 

Should we replace all the uses of this translationKey with a new one that'd take a parameter and deprecate the old one in this PR?

Reply:
We usually use the approach to follow best practices in new code without changing everything else. Now if this exact pattern is already used, I'm fine to keep it here.